src/base/bittorrent/peerinfo.cpp

```diff
@@ -181,6 +181,16 @@ QString PeerInfo::client() const
     return QString::fromStdString(m_nativeInfo.client);
 }
 
+QString PeerInfo::peerId() const
+{
+    // when peer ID is not known yet it contains only zero bytes,
+    // do not create string in such case, return empty string instead
+    if (*reinterpret_cast<const quint64*>(m_nativeInfo.pid.data()) == 0)
+        return {};
+    // peer ID in only first 8 bytes, the rest is not interesting
+    return QString::fromLatin1(m_nativeInfo.pid.data(), 8);
```

Comment:
@arvidn 
Is such an interpretation of the "peer is" correct? I see `peer_id` is just an alias for `sha1_hash`. So what is supposed to be used as "peer is" - a hash or something arbitrary? And what is actually used in the wild?

Reply:
https://wiki.theory.org/BitTorrentSpecification#peer_id

This implementation only caters to one encoding style.